archive/schema.go

```diff
@@ -0,0 +1,140 @@
+package archive
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/brimsec/zq/pkg/fs"
+)
+
+const configName = "zar.json"
+
+var DefaultConfig Config = Config{
+	Version:          0,
+	LogSizeThreshold: 500 * 1024 * 1024,
+}
+
+type Config struct {
+	Version          int `json:"version"`
+	LogSizeThreshold int `json:"log_size_threshold"`
+}
+
+func (c *Config) Write(path string) error {
+	b, err := json.Marshal(c)
+	if err != nil {
+		return err
+	}
+	f, err := ioutil.TempFile(filepath.Dir(path), "."+configName+".*")
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(b)
+	if err != nil {
+		f.Close()
```

Comment:
Being not quite sure myself how os' handle temp files, do you not need to delete the tempfile as well if there is an error?

Reply:
@mattnibs is right; we should close and then remove the temporary file on any error.

Reply:
👍